packages/core/src/modules/non-secret/service/NonSecretRecordService.ts

```diff
@@ -0,0 +1,40 @@
+import type { ConnectionRecord } from '../../connections/repository/ConnectionRecord'
+import type { NonSecretRecordTags } from '../repository/NonSecretRecord'
+
+import { Lifecycle, scoped } from 'tsyringe'
+
+import { AriesFrameworkError } from '../../../error'
+import { NonSecretRecord } from '../repository/NonSecretRecord'
+import { NonSecretRecordRepository } from '../repository/NonSecretRecordRepository'
+
+@scoped(Lifecycle.ContainerScoped)
+export class NonSecretRecordService {
+  private nonSecretRepository: NonSecretRecordRepository
+
+  public constructor(nonSecretRepository: NonSecretRecordRepository) {
+    this.nonSecretRepository = nonSecretRepository
+  }
+
+  public async saveRecord(message: string, tags?: NonSecretRecordTags, connectionRecord?: ConnectionRecord) {
+    const nonsecretRecord = new NonSecretRecord({
+      content: message,
+      connectionId: connectionRecord?.id,
+      tags: tags,
+    })
+
+    try {
+      await this.nonSecretRepository.save(nonsecretRecord)
+      return nonsecretRecord
+    } catch (error) {
+      throw new AriesFrameworkError(`${error}`)
+    }
+  }
+
+  public async findAllByQuery(query: Partial<NonSecretRecordTags>) {
```

Comment:
I think that we should keep this as `findByQuery` to keep the same name or we should rename all `findByQuery` to `findAllByQuery` I feel like `findAllByQuery` is a better name, as it is more explicit and `Array.find()` returns a single and not multiple so it can be confusing. Curious to here some opinions about this.

Reply:
I had this issue before! I think `findAllByQuery` is more clear. Also I had some confusions regarding the usage of 'find' and 'get' (e.g. get(record) will throw and exception if not found while getAll() returns empty array if no record found. Why not using findAll() in that case?)

Reply:
this function returns Array even if it is a single item so it should be findAll but any way the final word is for the maintainers

Reply:
Happy to change `getAll` to `findAll` for consistency, but probably better addressed in a separate PR. 

So we would use the following terms:

`findById` / `getById`
`findSingleByQuery` / `getSingleByQuery`
`findAll`
`findAllByQuery`

Anyone volunteering to address this in a separate PR? :)